modulus/models/diffusion/preconditioning.py

```diff
@@ -856,12 +856,21 @@ def __init__(
     def forward(
         self,
         x,
-        sigma,
-        class_labels=None,
         condition=None,
+        sigma=None,
+        class_labels=None,
         force_fp32=False,
         **model_kwargs,
     ):
+        # workaround to to handle backward compatibility with EDMPrecond, 
+        # argument ordering for EDMPrecond is x, condition, sigma
+        # EDMPrecond required condition to be passed, _ConditionalPrecond and 
+        # thus EDMPrecondV2 allows it to be None. To be backward compatible with
+        # EDMPrecond based code argument ordering has to be maintained which means
+        # sigma has to change from a positional only to a positional-or-keyword arg
+        if sigma == None:
+            raise ValueError("sigma must be provided, it cannot be None")
```

Comment:
I'm not happy with this fix, need to see if there's a better solution to this.

Reply:
Is `loss.py:336` the call site for this?

Reply:
That's the main location where the error occurs, it could also potentially occur at loss.py:602